model/map/src/main/java/org/keycloak/models/map/storage/ModelEntityUtil.java

```diff
@@ -154,6 +178,53 @@ public static <M> Class<M> getModelClass(String key) {
         return (Class<M>) NAME_TO_MODEL.get(key);
     }
 
+    @SuppressWarnings("unchecked")
+    public static boolean entityFieldsKnown(Class<?> entityClass) {
+        return ENTITY_FIELDS.containsKey(entityClass);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <V> Stream<EntityField<V>> getEntityFields(Class<V> entityClass) {
+        EntityField<V>[] values = (EntityField<V>[]) ENTITY_FIELDS.get(entityClass);
+        return values == null ? Stream.empty() : Stream.of(values);
+    }
+
+    public static <V extends AbstractEntity> Optional<EntityField<V>> getEntityField(Class<V> entityClass, String fieldNameCamelCase) {
+        final Stream<EntityField<V>> s = getEntityFields(entityClass);
+
+        return s
+          .filter(ef -> fieldNameCamelCase.equals(ef.getNameCamelCase()))
+          .findAny();
+    }
 
+    @SuppressWarnings("unchecked")
+    public static <V extends AbstractEntity> EntityField<V> getIdField(Class<V> targetEntityClass) {
```

Comment:
Unused method. Is it something for the future or it can be removed?

Reply:
It's something for the future, to be used later in the tree store.